include/library/spdm_common_lib.h

```diff
@@ -893,6 +893,26 @@ boolean libspdm_get_peer_cert_chain_data(IN void *spdm_context,
                       OUT void **cert_chain_data,
                       OUT uintn *cert_chain_data_size);
 
+/**
+  Retrieve the asymmetric public key from one DER-encoded X509 certificate,
+  based upon negotiated asymmetric or requester asymmetric algorithm.
+
+
+  @param  spdm_context                  A pointer to the SPDM context.
+  @param  algo                          SPDM base_asym_algo or req_base_asym_alg.
+  @param  cert_chain_data               Certitiface chain data without spdm_cert_chain_t header.
+  @param  cert_chain_data_size          size in bytes of the certitiface chain data.
+  @param  public_key                    Pointer to new-generated asymmetric context which contain the retrieved public key component.
+
+  @retval  TRUE   public key was retrieved successfully.
+  @retval  FALSE  Fail to retrieve public key from X509 certificate.
+**/
+boolean libspdm_get_public_key_from_cert_chain(IN void *context,
```

Comment:
1) Please move this function to spdm_crypt_lib.
2) Please change name to be libspdm_get_leaf_cert_public_key_from_cert_chain().
3) Please describe the expectation of the cert_chain - validated or not validated.